WcaOnRails/app/models/competition.rb

```diff
@@ -3,7 +3,7 @@
 class Competition < ApplicationRecord
   self.table_name = "Competitions"
 
-  has_many :competition_events, dependent: :destroy
+  has_many :competition_events, -> { order(:event_id) }, dependent: :destroy
```

Comment:
What issues did you run into trying to order them by `rank`? This works as expected for me (in rails console):
```rb
-> { joins(:event).order("Events.rank") }
```

Reply:
It's on the "through" associations that there is an issue:
```
irb(main):003:0> Competition.find("Euro2018").rounds
  Competition Load (0.4ms)  SELECT  `Competitions`.* FROM `Competitions` WHERE `Competitions`.`id` = 'Euro2018' LIMIT 1
  Round Load (5.0ms)  SELECT  `rounds`.* FROM `rounds` INNER JOIN `competition_events` ON `rounds`.`competition_event_id` = `competition_events`.`id` WHERE `competition_events`.`competition_id` = 'Euro2018' ORDER BY `rounds`.`number` ASC, Events.rank LIMIT 11
Traceback (most recent call last):
ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'Events.rank' in 'order clause': SELECT  `rounds`.* FROM `rounds` INNER JOIN `competition_events` ON `rounds`.`competition_event_id` = `competition_events`.`id` WHERE `competition_events`.`competition_id` = 'Euro2018' ORDER BY `rounds`.`number` ASC, Events.rank LIMIT 11)

```

Reply:
Right, I didn't check how `rounds` would behave. The `joins` gets lost somehow...

Reply:
Yes that's kind of unfortunate, and I couldn't figure out why :(